### ISSUE TITLE
Missing a CSS class of default style in block

### ISSUE THREAD

**Issue by dashkevych** [WordPress/gutenberg#40289]
### Description

Use case (example).

I have a global CSS styles for the Example block. They can be added by applying CSS styles to `.wp-block-example`.

In addition, I want to add styles just to a default block style. It can be done using `.wp-block-example.is-style-default`.

However, this class (`.is-style-default`) is not added to the block by default, although the Block editor indicates that a default style is selected.

An extra click on a default style in the Editor is needed in order to add the CSS class to the block.

### Step-by-step reproduction instructions

I am going to use the **Quote** block as an example. However, this issue can be applied to other blocks with multiple styles.

By default, the Quote block comes with multiple styles: Default and Plain (add Large).

1. Create a page.
2. Add the Quote block.
3. Notice the block style is automatically set to Default.
4. Publish the page.

<img width="1443" alt="step-1" src="https://user-images.githubusercontent.com/3655965/163091150-93a720db-6f7e-41ad-9e45-e0564d6a5c0e.png">

1. View the page in front-end.
2. Inspect the block to see its classes.
3. Notice that block has a base CSS class (`wp-block-quote`).

<img width="1313" alt="step-2" src="https://user-images.githubusercontent.com/3655965/163091391-92eb4578-46ef-4b01-ae7e-c48812eaffa2.png">

1. Go back to the editor.
2. Select the Quote block.
3. Click on the Default style in the sidebar.
4. Update the page.
5. Inspect the block in front-end.
6. Notice that additional CSS class (`is-style-default`) to the block.

<img width="1308" alt="step-3" src="https://user-images.githubusercontent.com/3655965/163091786-0832cf80-3547-463f-a7f3-d8d169589ea7.png">

### Environment info

- WordPress: 5.9.3.
- Theme: Twenty Twenty-Two
- No installed plugins.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Andrew-Starr** [comment]
FWIW I would like to see `is-style-default` removed completely.
Adding it by default would see every and all blocks carrying this extra class needlessly.

When clicking the Default style to remove a previously selected additional style, the `is-style-default` class should not be added to the markup IMO.

**Comment by houblon** [comment]
Confirming that it's required to click whatever block style is set as default to add the corresponding class to the block, including for custom block styles set as the default. Basically, setting a block style as default in code (PHP or JS) does nothing except falsely indicate that the block style is in effect.

Given the decision to remove the "Default Style" select menu for blocks (which did apply a corresponding CSS class to a block when it was created), it seems important to fix this in order to provide theme developers the ability to set default block styles. This is particularly the case for core blocks such as the Spacer block which have inline styles that must be overridden.

**Comment by fuzzywalrus** [comment]
// Edit
`isDefault` is the absence of CSS classes according to the docs.  Can't say I'm a fan of this without any option to have a preloaded class. It works fine for hand coding a block but cumbersome if simply registering block styles for `core` blocks or another developer's blocks.
https://developer.wordpress.org/block-editor/reference-guides/block-api/block-styles/
//




This also applies to custom blocks with additional styles. I've declared a custom style as the default for a block I'm coding, but the default style is not being applied to the block without clicking it. 

Here it is before a click:
<img width="2095" alt="Screen Shot 2022-07-08 at 10 44 50 AM" src="https://user-images.githubusercontent.com/2363648/178043920-a8ac6345-f9e8-4d6a-b521-50002f33e7ce.png">

And here it is after clicking the default style "bottom aligned", you'll see after the block name highlighted in yellow the `.is-style-bottom-aligned` class has now been applied.

<img width="2105" alt="Screen Shot 2022-07-08 at 10 47 40 AM" src="https://user-images.githubusercontent.com/2363648/178044196-f2f10290-3211-4e66-9451-910d5e1c7313.png">

This occurs if you use `wp.blocks.registerBlockStyle` or declare `styles: [ ]` with `isDefault: true` to any style, as they aren't written into the block,.

**Comment by frizzant** [comment]
is there a fix for this?

**Comment by frizzant** [comment]
I can do something like this to fix this issue:
 ```
registerBlockStyle('custom/example', {
        name: 'title-deco-1',
        label: 'Style 1',
        default: true,
        isDefault: true,
        attributes: {
            className: "is-style-title-deco-1"
        }
    });
```

**Comment by aristath** [comment]
A "default" style is not an actual style but the absence of a style. With that in mind, I don't think it makes sense to add a class for default styles... If it's a question of themes being able to target only the "default" style of a block, then that's possible in plain CSS using a selector like`.wp-block-something:not([class*="is-style-"])`.

**Comment by Ciantic** [comment]
@aristath but it still does add the class if you click the button with isDefault as true, it's just doesn't do it by default.

Removing the class is not usually wanted behavior. Imagine I have spacer with: small, medium and large. Where medium is default, then user adds 100 spacers all around the site, all without the style set because by default it doesn't. 

Then one day I want to change the default to small, should all the existing spacers jump suddenly to small? I doubt.

However maybe instead of "isDefault" there could be "isInitial", where isInitial chooses the initial value. IsDefault chooses the empty value.

**Comment by rocramer** [comment]
@aristath Fair enough. But then it is inconsistent to set the default class when you click explicitly on the default button. A consistent behaviour should be aimed. Everything else is confusing.

**Comment by sirreal** [comment]
Related to #38890.

**Comment by talldan** [comment]
> But then it is inconsistent to set the default class when you click explicitly on the default button. A consistent behaviour should be aimed. Everything else is confusing.

Completely agree with this, it'd be nice to fix that.

Also just to mention another factor for not adding a default class - a lot of users have blocks that are already 'default' and inserted in content that are missing the classname - it'd be very messy to retroactively add a default class.
